### DISCUSSION TITLE
TS 4058 error

### DISCUSSION THREAD

**Discussion by shlroland** [pmndrs/jotai#2758] [Q&A] [answered]
<img width="1249" alt="image" src="https://github.com/user-attachments/assets/db6ec3a6-7b44-48b3-a81d-22a96674e14d">

The reason for the issue in the `jotai` project might be the lack of exporting the `PrdStore` type.
In my project, the reason could be the activation of `composite: true`. Since I am working on a monorepo project, it is necessary to enable composite.
Do we need to consider the compatibility when `composite` is set to `true`?

related issues: #2522

**Comment by dai-shi** [comment, marked as answer]
also related: #2717

Does the workaround not work with `composite: true`?
https://tsplay.dev/w2eA8W
Otherwise, we need to do something.

**Reply by shlroland** [reply]
ok. Using code as shown in the example made ts error disappear.
